source/connection-troubleshooting.txt

```diff
@@ -0,0 +1,343 @@
+generates this error message:
+
+.. code-block:: none
+   :copyable: false
+
+   MongoServerSelectionError: connect ECONNRESET ::<IP address>:<port>
+
+The following section describes a method that may help resolve the issue.
+
+Increase the Number of File Descriptors
+---------------------------------------
+
+A file descriptor is a unique identifier associated with an open process. In most
+operating systems, each open connection from the driver is associated with a
+file descriptor. Operating systems typically have a limit on the number of file
+descriptors used by a single process. An ``ECONNRESET`` error can occur
+if the number of connections exceeds this limit.
+
+Consider the following example that creates a client and attempts to connect to
+a local deployment:
+
+.. literalinclude:: /code-snippets/faq/econnresetWithClientConnect-example.js
+  :language: javascript
+  :linenos:
+
+This operation can create a maximum number of 5000 connections as indicated by
+the inclusion of ``maxPoolSize`` on the URI. If the operating system has a file
+descriptor limit of 4000, this can cause an ``ECONNRESET`` error.
+
+To resolve this error, you can decrease the number of maximum allowed connections
+by setting the value of ``maxPoolSize``. Alternatively, you could increase the
+file descriptor limit in your operating system.
+
+.. warning::
+
+   Changing the configuration of your operating system should always be done
+   with caution.
+
+Authentication Error
+~~~~~~~~~~~~~~~~~~~~
+
+The {+driver-short+} can fail to connect to a MongoDB instance if
+the authorization is not configured correctly. If you are using ``SCRAM-SHA-256``
+for authentication and the driver fails to connect, the driver might raise an
+error message similar to one of the following messages:
+
+.. code-block:: none
+   :copyable: false
+
+   Command failed with error 18 (AuthenticationFailed): 'Authentication
+   failed.' on server <hostname>:<port>.
+
+.. code-block:: none
+   :copyable: false
+
+   connection() error occurred during connection handshake: auth error:
+   sasl conversation error: unable to authenticate using mechanism
+   "SCRAM-SHA-256": (AuthenticationFailed) Authentication failed.
+
+The following sections describe actions you can take to potentially resolve the
+issue.
+
+.. _node-troubleshooting-connection-string-auth:
+
+Check Your Connection String
+----------------------------
+
+An invalid connection string is the most common cause of authentication
+issues when attempting to connect to MongoDB using ``SCRAM-SHA-256``.
+
+.. tip::
+
+   For more information about connection strings,
+   see :ref:`Connection URI <node-connection-uri>` in the Connection Guide.
+
+If your connection string contains a username and password, ensure that they
+are in the correct format.
+
+.. note::
+
+   If the username or password includes any of the following characters, they
+   must be `percent encoded <https://tools.ietf.org/html/rfc3986#section-2.1>`__:
```

Comment:
Suggestion:

This seems like a distinct item to check rather than supplemental information that might go in an admonition. I think it would make sense to move the content outside of an admonition.

Reply:
I pulled it into the main text of the topic, because it feels like it fits strongly with verifying that the URI string is correct.